### DISCUSSION TITLE
Question about collection alter_alias error.

### DISCUSSION THREAD

**Discussion by hjlee9182** [milvus-io/milvus#38679] [Q&A and General discussion]
milvus version 2.3.13, pymilvus 2.3.7

Our server(kotlin) calls Milvus’s search by passing the parameter "a" for the collection, intending to query the collection associated with the alias "a."
Every day, as part of a daily batch process, the collection linked to the alias "a" is changed. The code structure looks roughly like the snippet below.

However, sometimes we observe that even though the alias has been updated, it still points to the old collection. Checking via Attu on the web also shows that the alias has indeed changed. Do you have any idea what might be causing this issue?
And this issue is occured sometimes not every day.

```

new_collection = create_collection(".....")
alter_alias( ".....", "a")
exist_collection = Collection("exsit...")
exist_collection.release()

```

**Comment by yhmo** [comment]
I could not reproduce the issue with this script, my milvus is v2.3.5
```
from pymilvus import (
    connections,
    FieldSchema, CollectionSchema, DataType,
    Collection,
    utility,
)

connections.connect(host='localhost', port=19530)
print(utility.get_server_version())

def create_collection(name, description):
    dim = 128
    fields = [
        FieldSchema(name="id", dtype=DataType.INT64, is_primary=True, auto_id=False),
        FieldSchema(name="vector", dtype=DataType.FLOAT_VECTOR, dim=dim),
    ]
    schema = CollectionSchema(fields, description=description)
    collection = Collection(name, schema)
    print("collection", name, "created")

ALIAS = "my_alias"
create_collection("test_0", "test_0_desc")
utility.create_alias("test_0", ALIAS)


next = 0
while True:
    old_name = "test_" + str(next)
    next = next + 1
    new_name = "test_" + str(next)
    description = new_name + "_desc"
    create_collection(name=new_name, description=description)
    utility.alter_alias(new_name, ALIAS)
    print("alias altered")
    collection = Collection(ALIAS)
    assert collection.description == description
    utility.drop_collection(old_name)
    print("collection", old_name, "dropped")

```

I repeated the process create_collection==>alter_alias==>check alias==>drop_collection for 500 times, I didn't get any problems.

**Comment by hjlee9182** [comment]
We didn't use a description, will this affect?

**Reply by yhmo** [reply]
No, I just use the description to certify that the Collection(ALIAS) is pointing to the new collection, not the old collection.
In your test code, how did you confirm that "it still points to the old collection."?

**Reply by hjlee9182** [reply]
That show real collection name. we check this solution

```

a = Collection( ALIAS )

a.describe()

```

**Reply by yhmo** [reply]
I just modified my test script and repeated the process for 500 times, the collection.describe() is correct, the desc["collection_name"] is always equal to the new collection name. Seems no problem.
```
    collection = Collection(ALIAS)
    desc = collection.describe()
    assert desc["collection_name"] == new_name
```

**Reply by hjlee9182** [reply]
Ah, the issue specifically arises when making calls from the Kotlin server to Milvus. The JDK version is 2.4.3. I recall seeing a past issue related to the collection cache—could it be related to that?

https://github.com/milvus-io/milvus/issues/25239

**Reply by yhmo** [reply]
Right, the Java SDK caches collection's info in MilvusClient for insert/upsert/delete interfaces, which could be the root cause. Currently, the cache cannot be updated manually. Let me thing about how to fix it.

To bypass this issue, you can declare a new MilvusClient each time before you create a new collection.

**Reply by hjlee9182** [reply]
Even if I change the schema (for example, modifying the description), the cache won't be updated, right?

**Reply by yhmo** [reply]
> Even if I change the schema (for example, modifying the description), the cache won't be updated, right?

Yes, the cache won't be updated.
Only if any insert/delete/upsert request fails, the cached schema is removed and the next insert/delete/upsert request will cache it again.
